src/net/sourceforge/kolmafia/swingui/panel/CommandDisplayPanel.java

```diff
@@ -56,6 +65,24 @@ public CommandDisplayPanel() {
     this.addFocusListener(this);
   }
 
+  private void loadHistoryFromPreference() {
+    String pref = getString(preference);
+    String[] commands = pref.split(DELIMITER);
+    if (!commands[0].trim().isEmpty()) {
+      Collections.addAll(commandHistory, commands);
+      this.commandIndex = commandHistory.size();
+    }
+  }
+
+  private void updatePreference() {
+    StringBuilder newPref = new StringBuilder();
+    for (String s : commandHistory) {
+      newPref.append(s).append(DELIMITER);
+    }
+    newPref.deleteCharAt(newPref.length() - 1);
+    setString(preference, newPref.toString());
```

Comment:
should be able to use `String.join(DELIMITER, commandHistory)` to avoid the for loop?

Reply:
Thank you.  I didn't remember/think of that.  Stand by.